src/osd/PeeringState.cc

```diff
@@ -3046,6 +3046,8 @@ void PeeringState::proc_primary_info(
     dirty_info = true;
     dirty_big_info = true;
   }
+  info.stats.stats.sum = oinfo.stats.stats.sum;
+  pl->publish_stats_to_osd();
```

Comment:
Not sure I understand how publish_stats_to_osd(), which starts with

void PG::publish_stats_to_osd()
{
  if (!is_primary())
    return;

has any effect when called after line 3031 above

Reply:
You are right, we don't need to publish it immediately(as publishing works only for primary osd). So whenever primary osd is down and secondary osd is taking control over at that time it will automatically publish the stats from the `recovery_state`. I did some adjustment in the code, such that the old codes are intact.